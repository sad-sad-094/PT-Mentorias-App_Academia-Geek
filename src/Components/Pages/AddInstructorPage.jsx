/* Author: Sebastian Aguirre Duque
E-mail: sadw621@gmail.com */

import { doc, setDoc } from 'firebase/firestore';
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import { BackButton, Button, ContainerHelper, Form, HContainer, Input, MainCard, MainContainer } from '../../Styles/GlobalStyles';
import { db } from '../../Utils/Firebase';
import NavbarUser from '../Modules/Navbar';

function AddInstructorPage() {

  const navigation = useNavigate();

  const defaultInstructor = () => {
    return {
      name: '',
      lastname: '',
      program: '',
      semester: '',
      DNI: '',
      number: '',
      email: ''
    }
  }

  const [instructor, setInstructor] = useState(defaultInstructor());

  const getInstructor = (event) => {
    setInstructor({
      ...instructor,
      [event.target.name]: event.target.value,
    })
  }

  const newInstructor = () => {
    setDoc(doc(db, 'MonitoresData', instructor.DNI), {
      name: instructor.name,
      lastname: instructor.lastname,
      program: instructor.program,
      semester: instructor.semester,
      DNI: instructor.DNI,
      number: instructor.number,
      email: instructor.email
    })
      .then(() => {
        toast.success('Instructor record saved successfully.')
      })
  }

  return (

    <MainContainer>

      <NavbarUser />

      <ContainerHelper>

        <MainCard>

          <h1>Add a new member to your team</h1>

          <Form onChange={getInstructor}>
            <Input type="text" placeholder="Please, enter a name" name="name" required />
            <Input type="text" placeholder="Please, enter a lastname" name="lastname" required />
            <Input type="text" placeholder="Please, enter a program" name="program" required />
            <Input type="text" placeholder="Please, enter a semester" name="semester" required />
            <Input type="number" placeholder="Please, enter a DNI" name="DNI" required />
            <Input type="number" placeholder="Please, enter a contact number" name="number" required />
            <Input type="email" placeholder="Please, enter a contact email" name="email" required />
          </Form>

          <HContainer>
            <Button onClick={newInstructor}>Submit</Button>
            <BackButton onClick={() => navigation("/userhome")}>Back</BackButton>
          </HContainer>

        </MainCard>

      </ContainerHelper>

    </MainContainer>

  )

}

export default AddInstructorPage;
