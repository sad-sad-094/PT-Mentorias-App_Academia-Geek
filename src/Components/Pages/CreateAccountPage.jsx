/* Author: Sebastian Aguirre Duque
E-mail: sadw621@gmail.com */

import React from 'react';
import { useNavigate } from 'react-router-dom';
import { BackButton, Button, HContainer, Input, MainCard, MainContainer } from '../../Styles/GlobalStyles';

function CreateAccountPage() {

  const navigation = useNavigate();

  return (

    <MainContainer>

      <MainCard>

        <h3>Welcome</h3>
        <p>to our app to manage your institution's instructors and their classes agendas.</p>

        <Input type="text" placeholder="Please, enter your full name" value="" required />
        <Input type="email" placeholder="Please, enter your email" value="" required />
        <Input type="password" placeholder="Please, enter your password" value="" required />
        <Input type="text" placeholder="Please, enter your faculty" value="" required />

        <HContainer>
          <Button>Signin</Button>
          <BackButton onClick={() => navigation("/") }>Back</BackButton>
        </HContainer>

      </MainCard>

    </MainContainer>
  )

}

export default CreateAccountPage;
