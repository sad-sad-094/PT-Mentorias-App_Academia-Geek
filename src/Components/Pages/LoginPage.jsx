/* Author: Sebastian Aguirre Duque
E-mail: sadw621@gmail.com */

import React from 'react';
import { useNavigate } from 'react-router-dom';
import { BackButton, Button, HContainer, Input, MainCard, MainContainer } from '../../Styles/GlobalStyles';

function LoginPage() {

  const navigation = useNavigate();

  return (

    <MainContainer>

      <MainCard>

        <h3>Welcome back manager</h3>
        <p>Return to your duties about your instructors' agenda.</p>

        <Input type="email" placeholder="Please, enter your email" value="" required />
        <Input type="password" placeholder="Please, enter your password" value="" required />

        <HContainer>
          <Button>Login</Button>
          <BackButton onClick={() => {navigation("/")}}>Back</BackButton>
        </HContainer>

      </MainCard>

    </MainContainer>
  )

}

export default LoginPage;
