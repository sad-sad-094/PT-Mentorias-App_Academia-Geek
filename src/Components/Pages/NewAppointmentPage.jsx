/* Author: Sebastian Aguirre Duque
E-mail: sadw621@gmail.com */

import React from 'react';
import { useNavigate } from 'react-router-dom';
import { BackButton, Button, HContainer, Input, MainCard, MainContainer } from '../../Styles/GlobalStyles';

function NewAppointmentPage() {

  const navigation = useNavigate();

  return (

    <MainContainer>

      <MainCard>

        <h3>Add a new appointment</h3>
        <p>Set a new appointment to your team.</p>

        <Input type="text" placeholder="Please, assign a date" name="date" required />
        <Input type="text" placeholder="Please, assign a course" name="course" required />
        <Input type="text" placeholder="Please, assign a classroom" name="classroom" required />
        <Input type="text" placeholder="Please, assign a instructor" name="instructor" required />

        <HContainer>
          <Button>Submit</Button>
          <BackButton onClick={() => navigation("/instructors")}>Back</BackButton>
        </HContainer>

      </MainCard>

    </MainContainer>

  )

}

export default NewAppointmentPage;
