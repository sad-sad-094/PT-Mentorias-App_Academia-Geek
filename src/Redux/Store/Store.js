/* Author: Sebastian Aguirre Duque
E-mail: sadw621@gmail.com */


import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import { logInReducer } from '../Reducers/Reducers';


const r = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose  //*Important Eliminar previo a realizar el deploy

const Store = createStore(
  combineReducers(
    {
      userLogIn: logInReducer,
    }
  ), r(applyMiddleware())
)

export default Store;
