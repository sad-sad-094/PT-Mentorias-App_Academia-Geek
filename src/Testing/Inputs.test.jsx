/* Author: Sebastian Aguirre Duque
E-mail: sadw621@gmail.com */

import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import AddInstructorPage from '../Components/Pages/AddInstructorPage';
import CreateAccountPage from '../Components/Pages/CreateAccountPage';
import LoginPage from '../Components/Pages/LoginPage';
import NewAppointmentPage from '../Components/Pages/NewAppointmentPage';
import Store from '../Redux/Store/Store';

describe('Inputs snapshots', () => {

  test('LogIn input snapshot', () => {
    const { container } = render(<Provider store={Store}> <BrowserRouter> <LoginPage /> </BrowserRouter> </Provider>);
    expect(container).toMatchSnapshot();
  });

  test('SingIn input snapshot', () => {
    const { container } = render(<Provider store={Store}> <BrowserRouter> <CreateAccountPage /> </BrowserRouter> </Provider>);
    expect(container).toMatchSnapshot();
  });

  test('Add instructors inputs snapshot', () => {
    const { container } = render(<Provider store={Store}> <BrowserRouter> <AddInstructorPage /> </BrowserRouter> </Provider>);
    expect(container).toMatchSnapshot();
  });

  test('Add new appointments inputs snapshot', () => {
    const { container } = render(<Provider store={Store}> <BrowserRouter> <NewAppointmentPage /> </BrowserRouter> </Provider>);
    expect(container).toMatchSnapshot();
  });

});

describe('Click event on buttons', () => {

  test('Click event on LogIn button', () => {

    render( <Provider store={Store}> <BrowserRouter> <LoginPage /> </BrowserRouter> </Provider> );
    fireEvent.click(screen.getByText('LogIn'));
    expect(screen.getByText('LogIn')).toBeTruthy();

  });

  test('Click event on SingIn button', () => {

    render(<Provider store={Store}> <BrowserRouter> <CreateAccountPage /> </BrowserRouter> </Provider>);
    fireEvent.click(screen.getByText('SignIn'));
    expect(screen.getByText('SignIn')).toBeTruthy();

  });

});
