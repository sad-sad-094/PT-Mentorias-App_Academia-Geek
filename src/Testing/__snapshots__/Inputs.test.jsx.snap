// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs snapshots Add instructors inputs snapshot 1`] = `
<div>
   
   
  <div
    class="sc-bczRLJ iIhikF"
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
          style="cursor: pointer;"
        >
          Monitor√≠as App
        </span>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              <img
                class="sc-iBkjds eozVut"
                src="https://res.cloudinary.com/dtxqusdhr/image/upload/v1661377910/PT-Monitorias-App/calendar_icon-icons.com_66541_wwmk92.png"
              />
            </a>
          </div>
          <div
            class="navbar-nav"
          >
            <a
              class="nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              Home
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="2"
              href="#"
              role="button"
              tabindex="0"
            >
              Logout
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="sc-gsnTZi jyHNcG"
    >
      <div
        class="sc-dkzDqf cswxdw"
      >
        <h1>
          Add a new member to your team
        </h1>
        <form
          class="sc-papXJ fXsUKo"
        >
          <input
            class="sc-ftvSup diFGBT"
            name="name"
            placeholder="Please, enter a name"
            required=""
            type="text"
          />
          <input
            class="sc-ftvSup diFGBT"
            name="lastname"
            placeholder="Please, enter a lastname"
            required=""
            type="text"
          />
          <input
            class="sc-ftvSup diFGBT"
            name="program"
            placeholder="Please, enter a program"
            required=""
            type="text"
          />
          <input
            class="sc-ftvSup diFGBT"
            name="semester"
            placeholder="Please, enter a semester"
            required=""
            type="text"
          />
          <input
            class="sc-ftvSup diFGBT"
            name="DNI"
            placeholder="Please, enter a DNI"
            required=""
            type="number"
          />
          <input
            class="sc-ftvSup diFGBT"
            name="number"
            placeholder="Please, enter a contact number"
            required=""
            type="number"
          />
          <input
            class="sc-ftvSup diFGBT"
            name="email"
            placeholder="Please, enter a contact email"
            required=""
            type="email"
          />
        </form>
        <div
          class="sc-hKMtZM greIcA"
        >
          <button
            class="sc-eCYdqJ ldjlBt"
          >
            Submit
          </button>
          <button
            class="sc-jSMfEi dUgSlV"
          >
            Back
          </button>
        </div>
      </div>
    </div>
  </div>
   
   
</div>
`;

exports[`Inputs snapshots Add new appointments inputs snapshot 1`] = `
<div>
   
   
  <div
    class="sc-bczRLJ iIhikF"
  >
    <div
      class="sc-dkzDqf cswxdw"
    >
      <h3>
        Add a new appointment
      </h3>
      <p>
        Set a new appointment to your team.
      </p>
      <form
        class="sc-papXJ fXsUKo"
      >
        <input
          class="sc-ftvSup diFGBT"
          name="date"
          placeholder="Please, assign a date"
          required=""
          type="text"
        />
        <input
          class="sc-ftvSup diFGBT"
          name="course"
          placeholder="Please, assign a course"
          required=""
          type="text"
        />
        <input
          class="sc-ftvSup diFGBT"
          name="classroom"
          placeholder="Please, assign a classroom"
          required=""
          type="text"
        />
      </form>
      <div
        class="sc-hKMtZM greIcA"
      >
        <button
          class="sc-eCYdqJ ldjlBt"
        >
          Submit
        </button>
        <button
          class="sc-jSMfEi dUgSlV"
        >
          Back
        </button>
      </div>
    </div>
  </div>
   
   
</div>
`;

exports[`Inputs snapshots LogIn input snapshot 1`] = `
<div>
   
   
  <div
    class="sc-bczRLJ iIhikF"
  >
    <div
      class="sc-dkzDqf cswxdw"
    >
      <h3>
        Welcome back manager
      </h3>
      <p>
        Return to your duties about your instructors' agenda.
      </p>
      <form
        class="sc-papXJ fXsUKo"
      >
        <input
          class="sc-ftvSup diFGBT"
          name="email"
          placeholder="Please, enter your email"
          required=""
          type="email"
        />
        <input
          class="sc-ftvSup diFGBT"
          name="password"
          placeholder="Please, enter your password"
          required=""
          type="password"
        />
      </form>
      <div
        class="sc-hKMtZM greIcA"
      >
        <button
          class="sc-eCYdqJ ldjlBt"
        >
          LogIn
        </button>
        <button
          class="sc-jSMfEi dUgSlV"
        >
          Back
        </button>
      </div>
      <p>
        Or LogIn with: 
        <svg
          enable-background="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="cursor: pointer;"
          version="1.1"
          viewBox="0 0 48 48"
          width="1em"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
            fill="#FFC107"
          />
          <path
            d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
            fill="#FF3D00"
          />
          <path
            d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
            fill="#4CAF50"
          />
          <path
            d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
            fill="#1976D2"
          />
        </svg>
         
      </p>
    </div>
  </div>
   
   
</div>
`;

exports[`Inputs snapshots SingIn input snapshot 1`] = `
<div>
   
   
  <div
    class="sc-bczRLJ iIhikF"
  >
    <div
      class="sc-dkzDqf cswxdw"
    >
      <h3>
        Welcome
      </h3>
      <p>
        to our app to manage your institution's instructors and their classes agendas.
      </p>
      <form
        class="sc-papXJ fXsUKo"
      >
        <input
          class="sc-ftvSup diFGBT"
          name="name"
          placeholder="Please, enter your full name"
          required=""
          type="text"
        />
        <input
          class="sc-ftvSup diFGBT"
          name="email"
          placeholder="Please, enter your email"
          required=""
          type="email"
        />
        <input
          class="sc-ftvSup diFGBT"
          name="password"
          placeholder="Please, enter your password"
          required=""
          type="password"
        />
        <input
          class="sc-ftvSup diFGBT"
          name="faculty"
          placeholder="Please, enter your faculty"
          required=""
          type="text"
        />
      </form>
      <div
        class="sc-hKMtZM greIcA"
      >
        <button
          class="sc-eCYdqJ ldjlBt"
        >
          SignIn
        </button>
        <button
          class="sc-jSMfEi dUgSlV"
        >
          Back
        </button>
      </div>
    </div>
  </div>
   
   
</div>
`;
